Assigning "a" + b() to int variable panics rather than complaining about type

The following program `sample.go` triggers a panic:

```go
package main

func b() string {
        return "b"
}

func main() {
        var x int
        x = "a" + b()
}
```

Expected result:
```console
$ go run ./sample.go
./sample.go:9:4: cannot use "a" + b() (type string) as type int in assignment
```

Got:
```console
$ yaegi ./sample.go
./sample.go:9:6: panic
run: runtime error: invalid memory address or nil pointer dereference
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc0002e0a00, 0x40)
        /usr/local/go/src/runtime/debug/stack.go:24 +0x9f
github.com/traefik/yaegi/interp.(*Interpreter).eval.func1(0xc0001b3c68)
        /go/src/yaegi/interp/interp.go:494 +0xc8
panic(0xe654a0, 0x158a780)
        /usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/traefik/yaegi/interp.runCfg.func1(0xc0003620b0, 0xc00046b500, 0xc0001b3a48)
        /go/src/yaegi/interp/run.go:185 +0x253
panic(0xe654a0, 0x158a780)
        /usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/traefik/yaegi/interp.add.func6(0xc0003620b0, 0xc000448930)
        /go/src/yaegi/interp/op.go:67 +0x40
github.com/traefik/yaegi/interp.runCfg(0xc00046b500, 0xc0003620b0)
        /go/src/yaegi/interp/run.go:191 +0x87
github.com/traefik/yaegi/interp.(*Interpreter).run(0xc000360000, 0xc00046ad00, 0xc000362000)
        /go/src/yaegi/interp/run.go:122 +0x2b0
github.com/traefik/yaegi/interp.(*Interpreter).eval(0xc000360000, 0xc0002b5560, 0x59, 0x7fff97a23efd, 0xb, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/yaegi/interp/interp.go:586 +0x80c
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0xc000360000, 0x7fff97a23efd, 0xb, 0xc0002b5500, 0x59, 0x0, 0x0, 0xc000210c30)
        /go/src/yaegi/interp/interp.go:408 +0x105
main.runFile(0xc000360000, 0x7fff97a23efd, 0xb, 0x1, 0x1)
        /go/src/yaegi/cmd/yaegi/run.go:123 +0xae
main.run(0xc000032050, 0x1, 0x1, 0xc00000a501, 0xc000001680)
        /go/src/yaegi/cmd/yaegi/run.go:89 +0x86e
main.main()
        /go/src/yaegi/cmd/yaegi/yaegi.go:144 +0x415
```

If I comment these lines:

```diff
diff --git a/interp/cfg.go b/interp/cfg.go
index 1301375..ab4b3a5 100644
--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -91,7 +91,7 @@ func (interp *Interpreter) cfg(root *node, importPath string) ([]*node, error) {
                                }
                                if !isInterface(dest.typ) {
                                        // Interface type are not propagated, and will be resolved at post-order.
-                                       n.typ = dest.typ
+                                       //n.typ = dest.typ
                                }
                        case binaryExpr, unaryExpr, parenExpr:
                                n.typ = n.anc.typ
@@ -731,13 +731,13 @@ func (interp *Interpreter) cfg(root *node, importPath string) ([]*node, error) {
                                // by constOp and available in n.rval. Nothing else to do at execution.
                                n.gen = nop
                                n.findex = notInFrame
-                       case n.anc.kind == assignStmt && n.anc.action == aAssign && n.anc.nleft == 1:
-                               // To avoid a copy in frame, if the result is to be assigned, store it directly
-                               // at the frame location of destination.
-                               dest := n.anc.child[childPos(n)-n.anc.nright]
-                               n.typ = dest.typ
-                               n.findex = dest.findex
-                               n.level = dest.level
+                       //case n.anc.kind == assignStmt && n.anc.action == aAssign && n.anc.nleft == 1:
+                       //      // To avoid a copy in frame, if the result is to be assigned, store it directly
+                       //      // at the frame location of destination.
+                       //      dest := n.anc.child[childPos(n)-n.anc.nright]
+                       //      n.typ = dest.typ
+                       //      n.findex = dest.findex
+                       //      n.level = dest.level
                        case n.anc.kind == returnStmt:
                                // To avoid a copy in frame, if the result is to be returned, store it directly
                                // at the frame location reserved for output arguments.
```

Then it works as expected (but presumably breaks something else):

```console
$ yaegi ./sample.go
run: ./sample.go:9:6: cannot use type string as type int in assignment
```

--- a/_test/issue-1093.go
+++ b/_test/issue-1093.go
@@ -1,0 +1,13 @@
+package main
+
+func b() string {
+	return "b"
+}
+
+func main() {
+	var x int
+	x = "a" + b()
+}
+
+// Error:
+// 9:6: cannot use type string as type int in assignment

--- a/interp/interp_consistent_test.go
+++ b/interp/interp_consistent_test.go
@@ -50,6 +50,7 @@ func TestInterpConsistencyBuild(t *testing.T) {
 			file.Name() == "import6.go" || // expect error
 			file.Name() == "init1.go" || // expect error
 			file.Name() == "io0.go" || // use random number
+			file.Name() == "issue-1093.go" || // expect error
 			file.Name() == "op1.go" || // expect error
 			file.Name() == "op7.go" || // expect error
 			file.Name() == "op9.go" || // expect error
@@ -208,6 +209,11 @@ func TestInterpErrorConsistency(t *testing.T) {
 			fileName:       "fun23.go",
 			expectedInterp: "3:17: too many arguments to return",
 			expectedExec:   "3:17: too many arguments to return",
+		},
+		{
+			fileName:       "issue-1093.go",
+			expectedInterp: "9:6: cannot use type string as type int in assignment",
+			expectedExec:   `9:4: cannot use "a" + b() (type string) as type int in assignment`,
 		},
 		{
 			fileName:       "op1.go",

--- a/interp/typecheck.go
+++ b/interp/typecheck.go
@@ -235,6 +235,15 @@ func (check typecheck) binaryExpr(n *node) error {
 	}
 
 	switch n.action {
+	case aAdd:
+		if n.typ == nil {
+			break
+		}
+		// Catch mixing string and number for "+" operator use.
+		k, k0, k1 := isNumber(n.typ.TypeOf()), isNumber(c0.typ.TypeOf()), isNumber(c1.typ.TypeOf())
+		if k != k0 || k != k1 {
+			return n.cfgErrorf("cannot use type %s as type %s in assignment", c0.typ.id(), n.typ.id())
+		}
 	case aRem:
 		if zeroConst(c1) {
 			return n.cfgErrorf("invalid operation: division by zero")

